dev-jjjjjeong-bin/Invaders-SDP
# Implementation of Player&EnemyShip Variety

# Bug Fixes and Boss Addition

### 1. Bug Fix: Enemy Ship Points Not Adding After Explosive Enemy Destruction
**What**  
This PR fixes a bug where enemy ships destroyed by an explosive enemy do not give points to the player.

**Why**  
Previously, points would only be added when an enemy ship was alive and destroyed directly by the player's bullet(s). Since explosive enemies destroy ships without them being hit by bullets, no points were being awarded. This fix ensures that points are correctly added even when ships are destroyed by explosions.

**Related Issue**  
Closes #14 of https://github.com/Dream-no24/PlayerEnemyShipVariety

**How to Test**  
1. Run the game.
2. Hit an explosive enemy with the player's bullet and observe whether points are added correctly.
3. Repeat this process on different levels to ensure consistency.

**Additional Information**  
- Fixed game freezing bug after destroying all ships. (84c62b7)
- Fixed bug where the explosion sprite disappeared too quickly during a chain explosion. (3dae4c2)
- Modified `_destroy(Bullet, EnemyShip)` to `_destroy(Bullet, EnemyShip, boolean)` in `EnemyShipFormation.java` to handle explosion logic more effectively.

### 2. Temporary Boss Addition on Level 3
**What**  
A temporary boss has been added to level 3.

**Why**  
Initially, I planned to exclude the boss enemy ship due to potential errors, as mentioned in a previous pull request to the TA repository. However, to align with the original game plan outlined in the `README.md` and to collaborate with the team working on boss-related features, a placeholder boss has been added to level 3.

---

This PR addresses critical gameplay bugs and includes a temporary boss for testing purposes. All changes have been internally tested, but further feedback or bug reports are welcome. Thank you for reviewing!
*Idea folder deleted from our repository.
